File: Sort Algorithms/Python/BubbleSort.py
class BubbleSort():
    def sort(self, List):
        Size=len(List)
        for i in range(Size-1):
          for j in range(Size-1):
              if List[j]>List[j+1]:                                      #If Element at next position is greater than current
                    temp=List[j]                                         #Then Swap both elements.
                    List[j]=List[j+1]
                    List[j+1]=temp
        return List

    def printArray(self, List):
        List = self.sort(List)
        for i in range(len(List)):
            print(List[i])
        return List

list = [5,4,3,2,1]
s = BubbleSort()
s.printArray(list) 

File: Sort Algorithms/Java/BubbleSort.java
import java.util.ArrayList;

public class BubbleSort<T extends Comparable<T>> extends SortAlgorithm<T> {
    @Override
    public void sort(ArrayList<T> arrayList) {
        boolean f = false;
        for (int i = 0; i < arrayList.size(); i++) {
            f=false;
            for (int j = 0; j < arrayList.size() - i - 1; j++) {
                if (arrayList.get(j).compareTo(arrayList.get(j + 1)) > 0) {
                    swap(arrayList, j, j + 1);
                    f = true;
                }
                if(!f)
                    break;
            }
        }
    }

    @Override
    public void sort(T[] array) {
        final int N = array.length;
        boolean f = false;
        for (int i = 0; i < N; i++) {
            f=false;
            for (int j = 0; j < N - i - 1; j++) {
                if (array[j].compareTo(array[j + 1]) > 0) {
                    swap(array, j, j + 1);
                    f=true;
                }
                if(!f)
                    break;
            }
        }
    }
}
